aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -840,7 +840,7 @@ Tensor compute_T12(const Tensor& A) {
     {num_prods, num_prods},
     {num_prods, 1},
     A.dtype()
-  ).to(A.device().type());
+  ).pin_memory().to(A.device().type(), /*non_blocking=*/true);
```

Comment:
I'm not sure this is doing what you want in the multi-gpu situation, you need `device_of` here

Reply:
Fixed. Thank you!